tests/test_browsing_context.py

```diff
@@ -985,3 +985,76 @@ async def test_browsingContext_navigationStarted_event(websocket, context_id,
             "url": url,
         }
     }
+
+
+@pytest.mark.asyncio
+async def test_browsingContext_userPromptOpened_event(websocket, context_id):
+
+    await subscribe(websocket, ["browsingContext.userPromptOpened"])
+
+    message = 'Prompt Opened'
+
+    await send_JSON_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": f"""alert('{message}')""",
+                "awaitPromise": True,
+                "target": {
+                    "context": context_id,
+                }
+            }
+        })
+
+    response = await read_JSON_message(websocket)
+    assert response == {
+        "method": "browsingContext.userPromptOpened",
+        "params": {
+            "context": context_id,
+            "type": 'alert',
+            "message": message,
+        }
+    }
+
+
+@pytest.mark.asyncio
+async def test_browsingContext_userPromptClosed_event(websocket, context_id):
+
+    await subscribe(websocket, [
+        "browsingContext.userPromptOpened", "browsingContext.userPromptClosed"
+    ])
+
+    message = 'Prompt Opened'
+
+    await send_JSON_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": f"""alert('{message}')""",
+                "awaitPromise": True,
+                "target": {
+                    "context": context_id,
+                }
+            }
+        })
+
+    await wait_for_event(websocket, "browsingContext.userPromptOpened")
```

Comment:
is it not flaky in Python? should not start waiting, then send_JSON_command, then await await wait_for_event?

Reply:
It reads the web socket till it finds the event with the correct name. So we SendCommand (not waiting for response) -> WaitForEvent to pop into the WebSocket response and then SendCommand ( again not awaiting )